notes/git-guidelines-commit-and-push.md

```diff
@@ -0,0 +1,74 @@
+## GUIDELINES TO GIT COMMIT AND PUSH
+
+Committing and pushing changes in Git involve several best practices to ensure a clean and organized version history. Here are some guidelines and best practices:
+
+### COMMITTING
+
+1. always check if you are in the right branch
+
+```bash
+    git branch
+```
+
+2. Add files with `git add .` (for all files) or `git add <file_name>` (for specific file)
+3. Check if the files you want are supposed to be staged with
+
+```bash
+git status
+```
+
+2. Review changes before committing with
+
+```bash
+    git diff
+```
+
+3. To check the branch name that HEAD is pointing to use
+
+```bash
+    git symbolic-ref —ref HEAD
+```
+
+### PUSHING
+
+1. Before pushing always check if you are on the correct branch
+
+```bash
+    git branch
+```
+
+2. Check if the branch name that HEAD is pointing to use
+
+```bash
+    git symbolic-ref —ref HEAD
+```
+
+3. Wound't hurt to double check the changes that you are about to push
+
+```bash
+    git diff
+```
+
+4. Always pull changes from the remote repository before pushing to avoid conflicts. To incorporate changes from the remote use
```

Comment:
well u are forgeting the most important  thing, to commit, otherwise u will push nothing

Reply:
pull and then commit